azurerm/internal/services/iotcentral/resource_arm_iotcentral_application.go

```diff
@@ -0,0 +1,257 @@
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving IoT Central Application %q (Resource Group %q):  %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil && *resp.ID != "" {
+		return fmt.Errorf("Error create IoT Central Application %q (Resource Group %q):  %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+	return resourceArmIotCentralAppRead(d, meta)
+}
+
+func resourceArmIotCentralAppRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).IoTCentral.AppsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["IoTApps"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving IoT Central Application %q (Resource Group %q):  %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if err := d.Set("sku", resp.Sku.Name); err != nil {
+		return fmt.Errorf("Error setting `sku`:  %+v", err)
+	}
+
+	d.Set("sub_domain", resp.AppProperties.Subdomain)
+	d.Set("display_name", resp.AppProperties.DisplayName)
+	d.Set("template", resp.AppProperties.Template)
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmIotCentralAppUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).IoTCentral.AppsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	displayName := d.Get("display_name").(string)
+	if displayName == "" {
+		displayName = name
+	}
+
+	subdomain := d.Get("sub_domain").(string)
+	template := d.Get("template").(string)
+	future, err := client.Update(ctx, resourceGroup, name, iotcentral.AppPatch{
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+		AppProperties: &iotcentral.AppProperties{
+			DisplayName: &displayName,
+			Subdomain:   &subdomain,
+			Template:    &template,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("Error update Iot Central Application %q (Resource Group %q).  %+v", name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for the completion of update Iot Central Application %q (Resource Group %q):  %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving IoT Central Application %q (Resource Group %q):  %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
```

Comment:
also need to judge whether it is empty, right?

Reply:
@xuzhang3  add   != "" check